src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -2096,7 +2194,32 @@ private void ValidateOrganizationUpgradeParameters(Models.StaticStore.Plan plan,
             upgrade.AdditionalSeats > plan.MaxAdditionalSeats.Value)
         {
             throw new BadRequestException($"Selected plan allows a maximum of " +
-                $"{plan.MaxAdditionalSeats.GetValueOrDefault(0)} additional users.");
+                                          $"{plan.MaxAdditionalSeats.GetValueOrDefault(0)} additional users.");
+        }
+    }
+
+    private static void ValidateSecretsManagerPlan(Models.StaticStore.Plan plan, OrganizationUpgrade upgrade)
+    {
+        ValidatePlan(plan, upgrade.AdditionalSmSeats, upgrade.PremiumAccessAddon, "Password Manager");
+
+        if (!plan.HasAdditionalServiceAccountOption && upgrade.AdditionalServiceAccount > 0)
```

Comment:
Please also validate that they do not have more SM seats than PM seats. Theoretically you could buy 10 PM seats and 20 SM seats, but you can never use those SM seats because all SM users must be PM users.

Reply:
added a check for the occupied seat count for secrets manager and password manager